src/web/client/utility/UrlBuilder.ts

```diff
@@ -3,34 +3,61 @@
+export function getRequestURL(
+    dataverseOrgUrl: string,
+    entity: string,
+    entityId: string,
+    method: string,
+    isSingleEntity: boolean,
+    attributeQueryParameters?: string): string {
     const powerPlatformContext = PowerPlatformExtensionContextManager.getPowerPlatformExtensionContext();
     let parameterizedUrlTemplate = getParameterizedRequestUrlTemplate(isSingleEntity);
+
     switch (method) {
         case httpMethod.GET:
             parameterizedUrlTemplate = parameterizedUrlTemplate
-                + powerPlatformContext.entitiesSchemaMap.get(pathParamToSchema.get(entity) as string)?.get('_fetchQueryParameters');
+                + (attributeQueryParameters ?? getEntity(entity)?.get(schemaEntityKey.FETCH_QUERY_PARAMETERS));
             break;
         default:
             break;
     }
 
-    return parameterizedUrlTemplate.replace('{dataverseOrgUrl}', dataverseOrgUrl).replace('{entity}', entity)
-        .replace('{entityId}', entityId).replace('{api}', powerPlatformContext.dataSourcePropertiesMap.get('api') as string)
-        .replace('{data}', powerPlatformContext.dataSourcePropertiesMap.get('data') as string)
-        .replace('{version}', powerPlatformContext.dataSourcePropertiesMap.get('version') as string);
+    return parameterizedUrlTemplate.replace('{dataverseOrgUrl}', dataverseOrgUrl).replace('{entity}', getEntity(entity)?.get(schemaEntityKey.DATAVERSE_ENTITY_NAME) as string)
+        .replace('{entityId}', entityId).replace('{api}', powerPlatformContext.dataSourcePropertiesMap.get(schemaKey.API) as string)
+        .replace('{data}', powerPlatformContext.dataSourcePropertiesMap.get(schemaKey.DATA) as string)
+        .replace('{version}', powerPlatformContext.dataSourcePropertiesMap.get(schemaKey.DATAVERSE_API_VERSION) as string);
+}
+
+export function getCustomRequestURL(dataverseOrgUrl: string, entity: string, urlQueryKey: string = schemaKey.MULTI_ENTITY_URL): string {
+    const powerPlatformContext = PowerPlatformExtensionContextManager.getPowerPlatformExtensionContext();
+    const parameterizedUrl = powerPlatformContext.dataSourcePropertiesMap.get(urlQueryKey) as string;
+    const fetchQueryParameters = getEntity(entity)?.get("_fetchQueryParameters");
+    const requestUrl = parameterizedUrl.replace('{dataverseOrgUrl}', dataverseOrgUrl)
+        .replace('{entity}', getEntity(entity)?.get(schemaEntityKey.DATAVERSE_ENTITY_NAME) as string)
+        .replace('{api}', powerPlatformContext.dataSourcePropertiesMap.get(schemaKey.API) as string)
+        .replace('{data}', powerPlatformContext.dataSourcePropertiesMap.get(schemaKey.DATA) as string)
+        .replace('{version}', powerPlatformContext.dataSourcePropertiesMap.get(schemaKey.DATAVERSE_API_VERSION) as string);
+
+    return requestUrl + fetchQueryParameters;
+}
+
+export function patchRequestUrl(entity: string, attributeType: string, requestUrl: string) {
```

Comment:
Can we rename this, it's returning a boolean, maybe isPatchRequest? Also can this be in utility?

Reply:
This is in utility files.